.github/workflows/hrm-lambdas-release.yml

```diff
@@ -0,0 +1,111 @@
+name: '[Multi Region] Deploy HRM Lambda'
+on:
+  workflow_dispatch:
+    inputs:
+      lambda_name:
+        description: 'hrm lambda name'
+        required: true
+        default: mock
+        type: choice
+        options:
+          - mock
+
+      environment:
+        description: HRM environment to deploy. E.G = development, staging, production (must match with the AWS environment value). Default value = development
+        default: development
+        required: true
+        type: choice
+        options:
+          - development
+          - staging
+          - production
+
+      #TODO: I've mapped region to environment, below. See note there about multi-region setup
+      # region:
+      #   description: AWS region to deploy. E.G = us-east-1, eu-west-1 (must match with the AWS environment value). Default value = us-east-1
+      #   default: us-east-1
+      #   required: true
+
+      send-slack-message:
+        description: 'Specifies if should send a Slack message at the end of successful run. Defaults to true'
+        required: true
+        type: boolean
+        default: true
+
+  workflow_call:
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      # Should probably use the passed in region for everything but the GITHUB_ACTIONS_SLACK_BOT_TOKEN SSM parameter only exists in us-east-1
+      AWS_DEFAULT_REGION:
+        required: true
+
+env:
+  AWS_DEFAULT_REGION: ${{ env.AWS_REGION }}
+  GITHUB_SHA: ${{ github.sha }}
+  DOCKER_IMAGE_BASE: 712893914485.dkr.ecr.${{ env.AWS_REGION }}.amazonaws.com/${{ inputs.environment }}-hrm-lambdas-
+
+jobs:
+  build-lambdas:
+    name: Build Lambdas
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+
+      # I still don't have full understanding of the region differences between env, but this is an option for skipping the region input
+      # if each env has multiple regions, this could be output and another matrix job could be produced to deploy to each region
+      - name: Set Env Based on Environment
+
+        run: |
+          case "${{ inputs.environment }}" in development) region="eu-west-1" ;; staging) region="eu-west-1" ;; production) region="eu-west-1" ;; *) echo "Invalid environment: ${{ inputs.environment }}" && exit 1 ;; esac &&
+          echo AWS_REGION=$region >> $GITHUB_ENV
```

Comment:
I'm not sure I get this chunk, what's the intention?

Reply:
This is an example of setting the env var AWS_REGION based on the environment instead of having to type/select two things. I still don't fully understand the region/env setup and could use some pointers on that. If an environment exists across regions, this could set an output to be used in a matrix workflow later to deploy to all regions for that environment without the dev knowing what they all are.

Reply:
Ah okay, I get it now. About the region-environment, our setup is like:
- region 1:
  - development
  - staging
  - production
- region 2:
  - development
  - staging
  - production

and so on. It would be ideal to preserve the ability to deploy to a particular region + environment combination, but having a "wrapper action" that "deploys to all staging" or something like that is also desirable in the short term.

Reply:
That makes sense. Right now I can only see a staging env in eu-west-1, but that may be a permissions issue.

Out of curiosity, what is a use case where one would want to deploy to development/staging/production in only one of the two regions?

Reply:
Just mentioned that we may want to get rid of that granularity in https://github.com/techmatters/infrastructure-config/pull/78#issuecomment-1259844008 :p

Reply:
We currently only have a staging HRM enviroment in eu-west-1, a prod environment is planned - we don't have any production helplines in that region yet though